### DISCUSSION TITLE
feature request: EdgeDB support

### DISCUSSION THREAD

**Discussion by notramo** [luckyframework/lucky#1720] [Feature requests & Ideas]
EdgeDB is a next-generation database.
Currently no Crystal web frameworks have integrated support for it unfortunately.
https://www.edgedb.com/

**Comment by jwoertink** [comment]
EdgeDB does look pretty cool. I think before any framework supports it, someone will need to write a shard adapter. With the next release of Lucky, it'll be a lot easier to swap out Avram for a different ORM, so if you wrote an EdgeDB ORM, you'd be able to use it.

**Comment by notramo** [comment]
Typical ORM mapping from 2 dimensional tables to objects is not needed as EdgeDB returns ready to use objects, but there are other aspects a database can be integrated to a framework.
I am currently looking into how to build an adapter for it.
